eZ/Publish/Core/MVC/Symfony/View/Tests/Builder/ContentViewBuilderTest.php

```diff
@@ -221,7 +224,89 @@ public function testBuildEmbedViewWithoutContentViewEmbedPermission(): void
         $this->contentViewBuilder->buildView($parameters);
     }
 
-    public function testBuildViewWithContentWhichDoesNotBelongsToLocation(): void
+    public function testBuildEmbedViewWithNullMainRequest(): void
+    {
+        $contentId = 120;
+        $content = new Content([
+            'versionInfo' => new VersionInfo([
+                'contentInfo' => new ContentInfo(['id' => $contentId]),
+                'status' => VersionInfo::STATUS_PUBLISHED,
+            ]),
+        ]);
+
+        $parameters = [
+            'viewType' => 'embed',
+            '_controller' => 'ez_content:embedAction',
+            'contentId' => $contentId,
+        ];
+
+        $this->requestStack
+            ->expects(self::once())
+            ->method('getMainRequest')
+            ->willReturn(null);
+
+        $this->repository
+            ->expects(self::once())
+            ->method('sudo')
+            ->willReturn($content);
+
+        $this->permissionResolver
+            ->expects(self::any())
+            ->method('canUser')
+            ->willReturn(true);
+
+        $this->contentViewBuilder->buildView($parameters);
+    }
+
+    public function testBuildEmbedViewWithNotNullMainRequest(): void
+    {
+        $contentId = 120;
+        $languageCode = 'ger-DE';
+
+        $content = new Content([
+            'versionInfo' => new VersionInfo([
+                'contentInfo' => new ContentInfo(['id' => $contentId]),
+                'status' => VersionInfo::STATUS_PUBLISHED,
+            ]),
+        ]);
+
+        $parameters = [
+            'viewType' => 'embed',
+            '_controller' => 'ez_content:embedAction',
+            'contentId' => $contentId,
+        ];
+
+        $attributes = $this->createMock(ParameterBag::class);
+        $attributes
+            ->expects(self::once())
+            ->method('get')
+            ->with('languageCode')
+            ->willReturn($languageCode);
+
+        $request = new Request();
+        $reflectionClass = new ReflectionClass($request);
+        $reflectionProperty = $reflectionClass->getProperty('attributes');
+        $reflectionProperty->setValue($request, $attributes);
```

Comment:
Why not simply:
```suggestion
        $request = new Request([], [], $attributes);
```
instead?

Reply:
Public property `$attributes` is actually a `parameterBag` which I mocked above. The idea was to test `$request->attributes->get('language')` invocation.

Reply:
You cannot test the output of the `buildView` method instead?

It seems awfully specific to test all those calls inside the service, and expect that fact alone to have an impact on the end result. Specifically, there is no guarantee based on these assertions that the language used for the content block rendered is the one that you will be passing.

You're only testing that services are called, not that whatever they return is used.

Reply:
> You cannot test the output of the `buildView` method instead?
> 
> It seems awfully specific to test all those calls inside the service, and expect that fact alone to have an impact on the end result. Specifically, there is no guarantee based on these assertions that the language used for the content block rendered is the one that you will be passing.
> 
> You're only testing that services are called, not that whatever they return is used.

Indeed we need to think here about better way of testing this, without a need for Reflection.

Reply:
Due to the fact that most objects are mocks and retrieving content takes place in a private method this might become challenging. Maybe some integration test should be composed in order to verify whether all the needed parameters are there?

Reply:
We decided within a team that proposed test coverage will do at the moment. We are lacking integration test coverage for that part of the system in general and implemented test cases are in-tact with how `ContentViewBuilder` is unit-tested for the time being.